### ISSUE TITLE
data lost when used in multiple cluster workers

### ISSUE THREAD

**Issue by xhawk18** [winstonjs/winston-daily-rotate-file#147]

**Comment by mattberther** [comment]
Please provide more information, including sample code and/or steps to reproduce the issue.

**Comment by bradisson** [comment]
I have a similar problem without using multiple cluster workers.
I set up winston-daily-rotate-file with the option maxSize to 10k (for test purpose) and my logger use also a console transporter.
When the first file reach the threshold, it rotate correctly but I found a delta between what is logged in the console and the concat of the two file.

simply put I get this result :

console output
```
[timestamp] - level: message 1
[timestamp] - level: message 2
[timestamp] - level: message 3
```

log1.log
```
[timestamp] - level: message 1
```

log2.log
```
[timestamp] - level: message 3
```

logger config
```
let transporters = [
  new transports.Console({
    timestamp: () => `[${(moment().format())}]`,
    colorize: true,
    json: false,
    level: config.application.log.level,
  }),
  new (require('winston-daily-rotate-file'))({
    timestamp: () => `[${(moment().format())}]`,
    datePattern: 'YYYY-MM-DD',
    level: config.application.log.level,
    filename: `/%DATE%.log`,
    dirname: join(process.cwd(), config.application.log.dir_all),
    maxSize: '10k',
    maxFiles: '3',
  }),
];

export const logger = new Logger({
  transports: transporters,
  exitOnError: false,
});
```

It seams that while the second file is build, some logs are lost.

**Comment by iWooli** [comment]
+1 for me
I have set 3 log files (debug, warn and error) and only debug are filled correctly. For warn and error logs, I have a delta between what's displayed on the console and what's written in the file.

Config
`var transport_debug = new (winston.transports.DailyRotateFile)({
    name: 'debug',
    dirname: 'logs',
    filename: 'debug-%DATE%.log',
    datePattern: 'YYYY-MM-DD-HH',
    level: 'debug',
    zippedArchive: false,
    json: true
});

var transport_warning = new (winston.transports.DailyRotateFile)({
    name: 'warning',
    dirname: 'logs',
    filename: 'warning-%DATE%.log',
    datePattern: 'YYYY-MM-DD-HH',
    level: 'warn',
    zippedArchive: false,
    json: true
});

var transport_error = new (winston.transports.DailyRotateFile)({
    name: 'error',
    dirname: 'logs',
    filename: 'error-%DATE%.log',
    datePattern: 'YYYY-MM-DD-HH',
    level: 'error',
    zippedArchive: false,
    json: true
});

var logger = new (winston.Logger)({
    transports: [
        transport_debug,
        transport_warning,
        transport_error
    ]
});`

Debug log file content
`{"level":"debug","message":"POST /controller/method { data: 1 } => []","timestamp":"2018-05-16T15:45:35.782Z"}
`
Error log file content
`{"level":"error","timestamp":"2018-05-16T15:40:00.086Z"}
{"level":"error","timestamp":"2018-05-16T15:42:26.736Z"}
{"level":"error","timestamp":"2018-05-16T15:42:26.738Z"}
{"level":"error","timestamp":"2018-05-16T15:42:26.742Z"}
{"level":"error","timestamp":"2018-05-16T15:45:30.915Z"}
{"level":"error","timestamp":"2018-05-16T15:45:30.916Z"}`

**Comment by mattberther** [comment]
@bradisson @iWooli I've identified a potential issue that may be causing this issue. Please see #156 for more information.

**Comment by mattberther** [comment]
Merged #156 and published to npm as `winston-daily-rotate-file@3.3.1`
